src/wallet/scriptpubkeyman.h

```diff
@@ -24,7 +46,23 @@ enum class OutputType;
  */
 class ScriptPubKeyMan
 {
+protected:
+    WalletStorage& m_storage;
+    std::shared_ptr<WalletDatabase> m_database;
+
 public:
+    ScriptPubKeyMan(WalletStorage& storage) : m_storage(storage), m_database(storage.GetDatabase()) {}
+
+    const std::string GetDisplayName() const { return m_storage.GetDisplayName(); }
```

Comment:
@meshcollider several of these (`protected`) methods are used by each scriptpubkeymanager. E.g. `IsWalletFlagSet()` is checked in `LegacyScriptPubKeyMan::UpgradeKeyMetadata()`. `UnsetWalletFlagWithDB()` is used to mark a wallet as no longer blank when a key is added. `GetDisplayName()` and `SetWalletFlag` are unused, but I'm fine with keeping them for completeness.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/16341#discussion_r338954347

> @meshcollider several of these (`protected`) methods are used by each scriptpubkeymanager. E.g. `IsWalletFlagSet()` is checked in `LegacyScriptPubKeyMan::UpgradeKeyMetadata()`. `UnsetWalletFlagWithDB()` is used to mark a wallet as no longer blank when a key is added. `GetDisplayName()` and `SetWalletFlag` are unused, but I'm fine with keeping them for completeness.

I think it would be best to eliminate **all** these methods because they obscure the difference between wallet and key manager objects and between storage and key management functionality. They also make call sites less clear about what the calls are intended to do.

In my [`pr/keyman` branch](https://github.com/ryanofsky/bitcoin/commits/pr/keyman) I've marked the change adding these methods (2d08c308a3d9936c389c86786e666c0b71c81052) as one of the questionable "Unclear if this change is necessary or useful" changes and put it at the end of the branch so it could be easily dropped or reverted.

Reply:
I don't have strong feelings on whether the left or right side of 2d08c30 is more clear. Explicit references to `m_storage` might discourage relying on it more; is the long term goal to get rid of `m_storage` usage in `ScriptPubKeyMan`s?

Reply:
> is the long term goal to get rid of `m_storage` usage in `ScriptPubKeyMan`s?

This would be a surprise to me. As far as I know, key managers will continue to need a way to read and store data in the wallet database, and `WalletStorage` provides a limited and controlled interface for doing that. 

Even if the wallet storage interface is going to be eliminated or cut back at some point, having wrapper functions that imply a key manager is storing its own data now when it isn't muddles the code and adds pointless hops in the meantime.

I don't think this is an important issue at all. Just surprised if getting rid of the indirection here wouldn't be the obvious choice.

Reply:
They can be removed in a follow up PR. The intention of wrapping up each function was to make the moves less painful and more clear that they were moves.